packages/file_selector/file_selector_web/lib/file_selector_web.dart

```diff
@@ -0,0 +1,222 @@
+import 'dart:async';
+import 'dart:html';
+import 'dart:typed_data';
+
+import 'package:file_selector_platform_interface/file_selector_platform_interface.dart';
+import 'package:flutter_web_plugins/flutter_web_plugins.dart';
+import 'package:meta/meta.dart';
+
+final String _kFileSelectorInputsDomId = '__file_selector_web_file_input';
+
+/// The web implementation of [FileSelectorPlatform].
+///
+/// This class implements the `package:file_selector` functionality for the web.
+class FileSelectorPlugin extends FileSelectorPlatform {
+  
+  /// Open file dialog for loading files and return a XFile
+  @override
+  Future<XFile> loadFile({
+    List<XTypeGroup> acceptedTypeGroups,
+    String initialDirectory,
+    String confirmButtonText,
+  }) {
```

Comment:
This function should probably be marked as `async`, so you can use `async/await` inside it, without worrying about having to use futures with `then` which is a little bit more cumbersome.